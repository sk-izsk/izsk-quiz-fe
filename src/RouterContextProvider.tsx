import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';

export interface RouterContextProviderProps {}

const RouterContextProvider: React.FC<RouterContextProviderProps> = ({ children }) => {
  const history: any = createMemoryHistory();
  return <Router history={history}>{children} </Router>;
};

export { RouterContextProvider };
