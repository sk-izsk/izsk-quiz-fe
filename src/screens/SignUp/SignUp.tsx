import { Box, makeStyles } from '@material-ui/core';
import React from 'react';
import { FcIdea } from 'react-icons/fc';
import { Link } from 'react-router-dom';
import { Button, CardAction, H6, TextField } from 'ui-neumorphism';
import { CardContainer } from '../../components';
import { CustomTheme, theme } from '../../theme/muiTheme';
import { useValueForTextField } from '../../utils';

export interface SignUpProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  mainContainer: {
    display: 'flex',
    width: '100vw',
    height: '100vh',
    justifyContent: 'center',
    alignItems: 'center',
  },
  inputContainer: {
    width: '100%',
    marginTop: theme.spacing(1),
    marginLeft: 0,
    marginRight: 0,
    marginBottom: 0,
  },
  cardContentStyle: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  btnContainer: {
    display: 'flex',
    justifyContent: 'space-evenly',
  },
  btn: {
    maxWidth: 200,
    width: '100%',
  },
  link: {
    textDecoration: 'none',
    width: '100%',
    color: theme.palette.primary.main,
    '&:hover': {
      textDecoration: 'none',
    },
  },
  header: {
    fontWeight: 'bold',
    color: theme.palette.primary.main,
  },
}));

const SignUp: React.FC<SignUpProps> = () => {
  const classes = useStyles();
  const [nickName, handleNickName] = useValueForTextField('');
  const [email, handleEmail] = useValueForTextField('');
  const [password, handlePassword] = useValueForTextField('');
  const [reTypePassword, handleReTypePassword] = useValueForTextField('');

  const handleSignUp = (event: KeyboardEvent) => {
    event.preventDefault();
    console.log(nickName, email, password, reTypePassword);
  };

  return (
    <Box className={classes.mainContainer}>
      <CardContainer
        cardContentStyle={classes.cardContentStyle}
        inset={true}
        cardAction={
          <CardAction className={classes.btnContainer}>
            <Link className={classes.link} to='/login'>
              <Button disabled={false} className={classes.btn} rounded color={theme.palette.primary.main}>
                Login
              </Button>
            </Link>
            <Button
              disabled={false}
              onClick={handleSignUp}
              className={classes.btn}
              rounded
              color={theme.palette.success.contrastText}
              bgColor={theme.palette.primary.main}
            >
              Sign Up
            </Button>
          </CardAction>
        }
      >
        <FcIdea size={70} />
        <H6 className={classes.header}>Ready to take the challenge !!!</H6>
        <TextField
          className={classes.inputContainer}
          width={400}
          placeholder='Enter your nick name'
          value={nickName}
          onChange={handleNickName}
          disabled={false}
          autofocus
        />
        <TextField
          className={classes.inputContainer}
          width={400}
          placeholder='Enter your email'
          value={email}
          onChange={handleEmail}
          disabled={false}
        />
        <TextField
          className={classes.inputContainer}
          width={400}
          placeholder='Enter your password'
          value={password}
          onChange={handlePassword}
          disabled={false}
          type='password'
        />
        <TextField
          className={classes.inputContainer}
          width={400}
          placeholder='Re type your password'
          value={reTypePassword}
          onChange={handleReTypePassword}
          disabled={false}
          type='password'
        />
      </CardContainer>
    </Box>
  );
};

export default SignUp;
