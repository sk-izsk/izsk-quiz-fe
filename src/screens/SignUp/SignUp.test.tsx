import { render } from '@testing-library/react';
import React from 'react';
import { RouterContextProvider } from '../../RouterContextProvider';
import SignUp from './SignUp';

test('SignUp renders correctly', () => {
  const { asFragment } = render(
    <RouterContextProvider>
      <SignUp />
    </RouterContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
