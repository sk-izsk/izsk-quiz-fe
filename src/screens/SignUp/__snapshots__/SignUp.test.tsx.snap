// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-8 makeStyles-mainContainer-1"
  >
    <div
      class="_28vIk _2bGnb _2Xmdp _1SFBz makeStyles-card-9"
    >
      <div
        class="_2So3v _2uw1B makeStyles-cardContentStyle-3"
      >
        <svg
          enable-background="new 0 0 48 48"
          fill="currentColor"
          height="70"
          size="70"
          stroke="currentColor"
          stroke-width="0"
          version="1"
          viewBox="0 0 48 48"
          width="70"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="24"
            cy="22"
            fill="#FFF59D"
            r="20"
          />
          <path
            d="M37,22c0-7.7-6.6-13.8-14.5-12.9c-6,0.7-10.8,5.5-11.4,11.5c-0.5,4.6,1.4,8.7,4.6,11.3 c1.4,1.2,2.3,2.9,2.3,4.8V37h12v-0.1c0-1.8,0.8-3.6,2.2-4.8C35.1,29.7,37,26.1,37,22z"
            fill="#FBC02D"
          />
          <path
            d="M30.6,20.2l-3-2c-0.3-0.2-0.8-0.2-1.1,0L24,19.8l-2.4-1.6c-0.3-0.2-0.8-0.2-1.1,0l-3,2 c-0.2,0.2-0.4,0.4-0.4,0.7s0,0.6,0.2,0.8l3.8,4.7V37h2V26c0-0.2-0.1-0.4-0.2-0.6l-3.3-4.1l1.5-1l2.4,1.6c0.3,0.2,0.8,0.2,1.1,0 l2.4-1.6l1.5,1l-3.3,4.1C25.1,25.6,25,25.8,25,26v11h2V26.4l3.8-4.7c0.2-0.2,0.3-0.5,0.2-0.8S30.8,20.3,30.6,20.2z"
            fill="#FFF59D"
          />
          <circle
            cx="24"
            cy="44"
            fill="#5C6BC0"
            r="3"
          />
          <path
            d="M26,45h-4c-2.2,0-4-1.8-4-4v-5h12v5C30,43.2,28.2,45,26,45z"
            fill="#9FA8DA"
          />
          <g
            fill="#5C6BC0"
          >
            <path
              d="M30,41l-11.6,1.6c0.3,0.7,0.9,1.4,1.6,1.8l9.4-1.3C29.8,42.5,30,41.8,30,41z"
            />
            <polygon
              points="18,38.7 18,40.7 30,39 30,37"
            />
          </g>
        </svg>
        <h6
          class="_3yPwv _2wZct _11JjB _1uvOH makeStyles-header-7"
        >
          Ready to take the challenge !!!
        </h6>
        <div
          class="_2xbD- makeStyles-inputContainer-2"
        >
          <div
            class="_fWLwS"
          >
            <input
              class="_U6nBC _2nHt_ "
              id="_50346_"
              placeholder="Enter your nick name"
              style="width: 400px; min-height: undefinedpx;"
              type="text"
              value=""
            />
            <div
              class="_3t8tT"
            >
              <div
                class="_3yPwv _2HbLl _1QRSN _1uvOH _2Lw0g"
              />
            </div>
          </div>
        </div>
        <div
          class="_2xbD- makeStyles-inputContainer-2"
        >
          <div
            class="_fWLwS"
          >
            <input
              class="_U6nBC _2nHt_ "
              id="_33100_"
              placeholder="Enter your email"
              style="width: 400px; min-height: undefinedpx;"
              type="text"
              value=""
            />
            <div
              class="_3t8tT"
            >
              <div
                class="_3yPwv _2HbLl _1QRSN _1uvOH _2Lw0g"
              />
            </div>
          </div>
        </div>
        <div
          class="_2xbD- makeStyles-inputContainer-2"
        >
          <div
            class="_fWLwS"
          >
            <input
              class="_U6nBC _2nHt_ "
              id="_67264_"
              placeholder="Enter your password"
              style="width: 400px; min-height: undefinedpx;"
              type="password"
              value=""
            />
            <div
              class="_3t8tT"
            >
              <div
                class="_3yPwv _2HbLl _1QRSN _1uvOH _2Lw0g"
              />
            </div>
          </div>
        </div>
        <div
          class="_2xbD- makeStyles-inputContainer-2"
        >
          <div
            class="_fWLwS"
          >
            <input
              class="_U6nBC _2nHt_ "
              id="_03513_"
              placeholder="Re type your password"
              style="width: 400px; min-height: undefinedpx;"
              type="password"
              value=""
            />
            <div
              class="_3t8tT"
            >
              <div
                class="_3yPwv _2HbLl _1QRSN _1uvOH _2Lw0g"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="_qCt7a _1l5Ig makeStyles-btnContainer-4"
      >
        <a
          class="makeStyles-link-6"
          href="/login"
        >
          <div
            class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND makeStyles-btn-5"
            style="color: rgb(33, 150, 243);"
          >
            <button
              class="_3lLt8"
            >
              Login
            </button>
          </div>
        </a>
        <div
          class="_cL4RU _17_OS  _3W7Om _CWiZp _39_ND makeStyles-btn-5"
          style="color: rgb(238, 252, 243); background-color: rgb(33, 150, 243);"
        >
          <button
            class="_3lLt8"
          >
            Sign Up
          </button>
        </div>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
